src/compas_timber/connections/l_halflap.py

```diff
@@ -0,0 +1,217 @@
+        return cfr
+
+    @property
+    def cutting_plane_cross(self):
+        angles_faces = beam_side_incidence(self.cross_beam, self.main_beam)
+        cfr = max(angles_faces, key=lambda x: x[0])[1]
+        return cfr
+
+    def _create_negative_volumes(self):
+        # Get Cut Plane
+        plane_cut_vector = self.main_beam.centerline.vector.cross(self.cross_beam.centerline.vector)
+
+        if self.flip_lap_side:
+            plane_cut_vector = -plane_cut_vector
+
+        # Get Beam Faces (Planes) in right order
+        planes_main = self._sort_beam_planes(self.main_beam, plane_cut_vector)
+        plane_a0, plane_a1, plane_a2, plane_a3 = planes_main
+
+        planes_cross = self._sort_beam_planes(self.cross_beam, -plane_cut_vector)
+        plane_b0, plane_b1, plane_b2, plane_b3 = planes_cross
+
+        # Lines as Frame Intersections
+        lines = []
+        unbound_line = intersection_plane_plane(plane_a1, plane_b1, tol=1e-6)
+
+        pt_a = intersection_line_plane(unbound_line, plane_a0)
+        pt_b = intersection_line_plane(unbound_line, plane_b0)
+        lines.append(Line(pt_a, pt_b))
+
+        unbound_line = intersection_plane_plane(plane_a1, plane_b2)
+        pt_a = intersection_line_plane(unbound_line, plane_a0)
+        pt_b = intersection_line_plane(unbound_line, plane_b0)
+        lines.append(Line(pt_a, pt_b))
+
+        unbound_line = intersection_plane_plane(plane_a2, plane_b2)
+        pt_a = intersection_line_plane(unbound_line, plane_a0)
+        pt_b = intersection_line_plane(unbound_line, plane_b0)
+        lines.append(Line(pt_a, pt_b))
+
+        unbound_line = intersection_plane_plane(plane_a2, plane_b1)
+        pt_a = intersection_line_plane(unbound_line, plane_a0)
+        pt_b = intersection_line_plane(unbound_line, plane_b0)
+        lines.append(Line(pt_a, pt_b))
+
+        # Create Polyhedrons
+        negative_polyhedron_main_beam = self._create_polyhedron(plane_a0, lines, self.cut_plane_bias)
+        negative_polyhedron_cross_beam = self._create_polyhedron(plane_b0, lines, self.cut_plane_bias)
+        return negative_polyhedron_main_beam, negative_polyhedron_cross_beam
+
+    def restore_beams_from_keys(self, assemly):
+        """After de-serialization, resotres references to the main and cross beams saved in the assembly."""
+        self.main_beam = assemly.find_by_key(self.main_beam_key)
+        self.cross_beam = assemly.find_by_key(self.cross_beam_key)
+
+    def add_features(self):
+        start_main, end_main = self.main_beam.extension_to_plane(self.cutting_plane_main)
+        start_cross, end_cross = self.cross_beam.extension_to_plane(self.cutting_plane_cross)
+        # self.main_beam.add_blank_extension(start_main, end_main, self.key)
+        # self.cross_beam.add_blank_extension(start_cross, end_cross, self.key)
+        self.main_beam.add_blank_extension(1, 1, self.key)
+        self.cross_beam.add_blank_extension(1, 1, self.key)
```

Comment:
are these hard-coded extensions intentional? shouldn't the by calculated somehow?

Reply:
same as l-butt cross-beam extension. returns error when trimming the beams extended with (start_main, end_main,..)

This is a complete non-solution because I model in mm. (BTLx, too). we should find a solution ASAP